tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/LazyEquality.scala

```diff
@@ -424,10 +424,17 @@ class LazyEquality(rewriter: SymbStateRewriter)
       case `arraysEncoding` =>
         // In the arrays encoding we only cache the equalities between the elements of the functions' ranges
         // This is because the ranges consist of pairs of form <arg,res>, thus the domains are also handled
+        // TODO: stop caching unnecessary equalities, similar to SetInRuleWithArrays.basicIn?
```

Comment:
This will be resolved in #1770. So it probably does not need additional treatment here.

Reply:
Good to know, thanks!